### DISCUSSION TITLE
Empty pages on redirect until page refreshed

### DISCUSSION THREAD

**Discussion by orafal-dev** [nuxt/nuxt#17408] [Questions]
Having issue with routing or fetching (that's what I'm trying to figure out).

Using Nuxt v3 rc3

Root page launches, and renders perfectly fine as it should.
Whenever I go to different subpage, for this example: `/blog` via `NuxtLink` component, the page renders blank (white page). This page actually has 3 `useAsyncData` doing fetches, tried them with `await` and without - nothing changes.

BUT!

When I refresh the same page when on the white, unrendered page - it fetches fine, launches the blog list page and it's fine.

Here's quick recording of this behaviour https://share.cleanshot.com/aEWiZj

Tried with `ssr: false` and `true` also `target: 'static'` and `server`, and without any of these options.

**Comment by orafal-dev** [comment]
bumping, at least let me know if it's an issue on my side or internal bug

**Comment by Diizzayy** [comment]
@rafalolszewski94 I suspect that the issue lies on your end, We'd be better able to help you resolve this if you provide code snippets or a reproduction of this behavior.

**Comment by GanpatJangra** [comment]


**Reply by danielroe** [reply]
When he suggested the _code_ he didn't mean a screenshot. Start here: https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz. I would also just confirm each of your pages and layouts has a single root node.

**Reply by GanpatJangra** [reply]
> When he suggested the _code_ he didn't mean a screenshot. Start here: https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz. I would also just confirm each of your pages and layouts has a single root node.

Yes
